mantid_release_note_check/mantid_release_note_check.py

```diff
@@ -0,0 +1,53 @@
+import argparse
+
+from pathlib import Path
+from typing import Sequence
+
+
+def filter_files(filenames: set[str]) -> set[str]:
+    files_to_keep = set()
+    for filename in filenames:
+        path = Path(filename)
+        if path.suffix == '.rst' and 'release' in path.parts:
+            files_to_keep.add(filename)
+
+    return filenames & files_to_keep
+
+
+def check_bullet_points(filenames: set[str]) -> int:
+    retv = 0
+
+    for filename in filenames:
+        with open(filename, 'r') as rn:
+            lines = rn.readlines()
+            for i in range(len(lines)):
+                line = lines[i].lstrip()
+                if len(lines[i]) - len(line) > 1 or len(line) == 0:
+                    # ignore nested bullet points and blank lines
+                    continue
+                bullet = line[0]
+                if bullet in ['*', '+']:
+                    print(f"{filename} has incorrect bullet point style '{bullet}'. Please only use '-'.")
+                    print(f"Fixing {filename}")
+                    lines[i] = lines[i].replace(bullet, '-', 1)
+                    retv = 1
+
+        with open(filename, 'w') as rn:
```

Comment:
It looks like this writes to the file even when the file has not changed. Is it possible to only write if a change was made?